ddtrace/tracer/textmap.go

```diff
@@ -114,74 +115,188 @@ func NewPropagator(cfg *PropagatorConfig) Propagator {
 	if cfg.PriorityHeader == "" {
 		cfg.PriorityHeader = DefaultPriorityHeader
 	}
-	return &propagator{cfg}
+	return &propagator{
+		injectors:  makeInjectors(cfg),
+		extractors: makeExtractors(cfg),
+	}
 }
 
-// propagator implements a propagator which uses TextMap internally.
+// propagator implements a Propagator that supports TextMap carriers.
 // It propagates the trace and span IDs, as well as the baggage from the
 // context.
-type propagator struct{ cfg *PropagatorConfig }
+type propagator struct {
+	injectors  []textMapPropagator
```

Comment:
Even though this is nice and specific, I don't think there is a need for this new interface. It complicates things a bit. Could we maybe just have:
```go
// chainPropagator chains a context through multiple injectors when injecting. When extracting, it tries
// multiple extractors selecting the first successful one.
type chainPropagator struct {
    injectors []Propagator
    extractors []Propagator
}
```
Then we can have two propagators:
* `propagator` - the regular Datadog propagator we had before
* `propagatorB3` - the B3 propagator

I think this would simplify things a bit both to the reader and to us. For example, if we want to implement binary B3 propagator sometime, this will make it much easier to add it in.